syntax = "proto3";

package remote_execution;

service RemoteExecutionService {
  rpc ExecuteOp(OpRequest) returns (OpResponse);
}

message TensorProto {
  bytes data = 1;
  repeated int64 shape = 2;
  string dtype = 3;
}

message OpRequest {
  string op_name = 1;
  string overload_name = 2;
  repeated TensorProto arguments = 3;
}

message OpResponse {
  TensorProto result = 1;
}

